### PR TITLE
Updated identifier-expression.js for issue #58

### PR DESCRIPTION
**Pull request by DylanLWScott**
Added die on reference being '_'

### CODE REVIEW

**Comment by wavebeem** [comment]
@DylanLWScott

### :tada: Thanks for the PR!  :tada:

The implementation looks good. :+1: 

I would like to avoid abbreviations in compiler output unless necessary. Do you think the following wording would make sense instead?

```
squiggle: variable '_' cannot be referenced as it will never be bound
```

I'll give it a test run once the wording seems ideal. :car:

**Comment by DylanLWScott** [comment]
Thanks for the suggestion!  I think it would be clearer.  I had some issues getting around the 80 character limit per line on the error message, so using the '+' operator.  However, please let me know if there is a better way.

**Comment by wavebeem** [comment]
I agree `+` was the right choice. I'll test this later then merge. Thanks for your help! :+1:

**Comment by wavebeem** [comment]
Just tested it and it's not working correctly:

```
🔥  ./bin/build.rb ; and echo '_' | squiggle /dev/stdin | node
squiggle: lint: undeclared variable _
[stdin]:378
    return $ref(_, '_');
                ^

ReferenceError: _ is not defined
```

I'll leave a line comment where I noticed the bug.

File: src/transform/identifier-expression.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
+var die = require("../die");
 var es = require("../es");
 
 function IdentifierExpression(transform, node) {
     var id = transform(node.data);
     var name = es.Literal(node.data.data);
     var ref = es.Identifier("$ref");
+    if (ref === "_") {
```
Comment (wavebeem) [inline src/transform/identifier-expression.js]: `ref` isn't the right thing to check for `"_"`. Add `console.error(ref)` and you can see what that object looks like and how  it doesn't vary.

Add a `console.error(node)` and you should see where the `"_"` will show up.

**Comment by DylanLWScott** [comment]
Sorry about that.  It was good to see how you invoked it in your comment->showing how to pass that value in as stdin was great!  Let me know if anything else needs to change and thanks for the help so far!

**Comment by wavebeem** [comment]
Huzzah! It seems to work great now. I did a little more testing and realized `die` was actually a bad idea on my part. This affects the behavior of the interactive mode (`squiggle -i`):

```
prompt$ squiggle -i
Welcome to Squiggle 0.3.0
Type :help for more information, or :quit to quit.

squiggle> _
squiggle: variable '_' cannot be referenced as it will never be bound

prompt$
```

I'm thinking this should throw an error now instead of calling `die()` so that the interactive mode can catch the problem. Would you be up for making that change?

**Comment by DylanLWScott** [comment]
Thanks the the mentoring!  I took a slightly different approach-what do you think about looking in die.js for if interactive mode is enabled?  If it is, don't exit the process, but if it is not, exit like normal?  I submitted that idea but am ok with changing it to be an error as you already suggested-sometimes simpler is better ;)

**Comment by wavebeem** [comment]
That could work, but I think it would be best to keep the command line options separate from the compiler internals. Also, this would still allow the code to compile in interactive mode, which doesn't make sense.

I think for now throwing an error is probably the best solution as the interactive mode and compiler modes could decide what to do themselves.

**Comment by DylanLWScott** [comment]
Updated to revert die.js change and to throw Error

**Comment by wavebeem** [comment]
Looks good and works well. Compiler mode should try to catch the error eventually, but it's good enough for now. Thanks!

Happy to help if any of the other issues interest you.

**Pull request merged**